WhateverGreen/kern_igfx.hpp

```diff
@@ -570,1062 +508,703 @@ class IGFX {
-		 *  @return A driver instance on success, `nullptr` otherwise.
-		 *  @note This convenient initializer returns `nullptr` if it could not retrieve the framebuffer index.
+		 *  @note This method will update fields in `lspcons` accordingly on success.
 		 */
-		static LSPCON *create(void *controller, IORegistryEntry *framebuffer, void *displayPath) {
-			// Guard: Framebuffer index should exist
-			uint32_t index;
-			if (!AppleIntelFramebufferExplorer::getIndex(framebuffer, index))
-				return nullptr;
-
-			// Call the private constructor
-			return new LSPCON(controller, framebuffer, displayPath, index);
-		}
+		void setupLSPCON(void *that, IORegistryEntry *framebuffer, void *displayPath);
 
 		/**
-		 *  [Convenient] Destroy the LSPCON driver safely
+		 *  [Convenient] Check whether the given framebuffer has an onboard LSPCON chip
 		 *
-		 *  @param instance A nullable LSPCON driver instance
+		 *  @param index A **valid** framebuffer index; Must be less than `MaxFramebufferConnectorCount`
+		 *  @return `true` if the framebuffer has an onboard LSPCON chip, `false` otherwise.
 		 */
-		static void deleter(LSPCON *instance NONNULL) { delete instance; }
-
-		/**
-		 *  Probe the onboard LSPCON chip
-		 *
-		 *  @return `kIOReturnSuccess` on success, errors otherwise
-		 */
-		IOReturn probe();
-
-		/**
-		 *  Get the current adapter mode
-		 *
-		 *  @param mode The current adapter mode on return.
-		 *  @return `kIOReturnSuccess` on success, errors otherwise.
-		 */
-		IOReturn getMode(Mode *mode);
-
-		/**
-		 *  Change the adapter mode
-		 *
-		 *  @param newMode The new adapter mode
-		 *  @return `kIOReturnSuccess` on success, errors otherwise.
-		 *  @note This method will not return until `newMode` is effective.
-		 *  @warning This method will return the result of the last attempt if timed out on waiting for `newMode` to be effective.
-		 */
-		IOReturn setMode(Mode newMode);
-
-		/**
-		 *  Change the adapter mode if necessary
-		 *
-		 *  @param newMode The new adapter mode
-		 *  @return `kIOReturnSuccess` on success, errors otherwise.
-		 *  @note This method is a wrapper of `setMode` and will only set the new mode if `newMode` is not currently effective.
-		 *  @seealso `setMode(newMode:)`
-		 */
-		IOReturn setModeIfNecessary(Mode newMode);
-
-		/**
-		 *  Wake up the native DisplayPort AUX channel for this adapter
-		 *
-		 *  @return `kIOReturnSuccess` on success, other errors otherwise.
-		 */
-		IOReturn wakeUpNativeAUX();
-
-		/**
-		 *  Return `true` if the adapter is running in the given mode
-		 *
-		 *  @param mode The expected mode; one of `LS` and `PCON`
-		 */
-		inline bool isRunningInMode(Mode mode) {
-			Mode currentMode;
-			if (getMode(&currentMode) != kIOReturnSuccess) {
-				DBGLOG("igfx", "LSPCON::isRunningInMode() Error: [FB%d] Failed to get the current adapter mode.", index);
-				return false;
-			}
-			return mode == currentMode;
+		inline bool hasLSPCON(uint32_t index) {
```

Comment:
When declaring a class function within the class you can safely omit `inline`, this is implied.